src/sinks/kafka/config.rs

```diff
@@ -20,12 +26,34 @@ use crate::{
 
 pub(crate) const QUEUED_MIN_MESSAGES: u64 = 100000;
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Migrator;
+
+impl EncodingConfigMigrator for Migrator {
+    type Codec = StandardEncodings;
+
+    fn migrate(codec: &Self::Codec) -> (Option<FramingConfig>, SerializerConfig) {
+        let framing = match codec {
+            StandardEncodings::Text | StandardEncodings::Json => None,
+            StandardEncodings::Ndjson => Some(NewlineDelimitedEncoderConfig::new().into()),
+        };
```

Comment:
Does having a framer in a message-oriented sink like kafka make sense?

Reply:
Yeah, I had this same thought. It seems very unlikely to me that someone would want to use anything other than message framing for any message-based sink (Kafka, UDP, AWS SQS, GCP Pub Sub, etc.). I'd support restricting it to message-based framing until we get a user with a use-case for something else.

Reply:
Agree, that makes more sense. Have a PR up for the necessary support structures: https://github.com/vectordotdev/vector/pull/12156.